### ISSUE TITLE
Warning: Facebook SDK 4.6 requires Android SDK >= 15

### ISSUE THREAD

**Issue by jeduan** [jeduan/cordova-plugin-facebook4#39]
Leaving this here until I update the docs. A workaround is adding a preference to `config.xml`

```xml
<platform name="android">
    <preference name="android-minSdkVersion" value="15" />
</platform>
```

**Comment by Hirbod** [comment]
This is very weird, as I have ```value="16"``` inside of my config.xml. Also tried 15, but I get

```
> Manifest merger failed : uses-sdk:minSdkVersion 14 cannot be smaller than version 15 declared in library /app/platforms/android/build/intermediates/exploded-aar/com.facebook.android/facebook-android-sdk/4.6.0/AndroidManifest.xml
```

**Comment by Hirbod** [comment]
This works, thanks. Now I've deleted my platform and forgot to save my keystore (which was inside of it). Brilliant :D

Thank god the app wasn't published.

**Comment by Hirbod** [comment]
I guess it has something to do with Crosswalk

**Comment by Justin-Credible** [comment]
I have the same issue; using @jeduan's suggestion of specifying the minimum SDK version on the command line worked, but I'm still curious as to why it is needed.

**Comment by KeKs0r** [comment]
Having the same issue and in my config I have minSdkVersion=21. Also Tried to run the command you provided. Still same error message

**Comment by KeKs0r** [comment]
I think I found a workaround, or at least where it is coming from. @Hirbod was right. Its coming from crosswalk.

Just check android-xwalk.cradle and there is the following line:

```
if (cdvMinSdkVersion == null) {
    ext.cdvMinSdkVersion = 14
}
```

I don't know why crosswalk assumes othat the MinSdkVersion is null, but if you change that 14, to a 15. at least you can build.

**Comment by fredgalvao** [comment]
@KeKs0r  that code only runs if you don't specify a minSdkVersion on the command line, so if you do like @jeduan suggested it will work. And yes, that's related to crosswalk's dependencies, not to facebook's.

`cordova build android --device --release -- --minSdkVersion=15`

(I'm almost sure the `--device` flag above does nothing, as the android build is the same for emulator and device, even if it's a crosswalk build)

**Comment by tehmaestro** [comment]
Hi. I want to support sdk 14, so I forked this repository and added a fixed facebook.sdk 4.5.0 in `plugin.xml`. However, when calling `cordova build` I get two dependencies added, `4.5.0` and `4+` which of course fails.

  Do I need to modify anything else in order to fix the sdk version to `4.5.0` ?

**Comment by LukeeeeBennett** [comment]
Would someone be kind enough to elaborate on the mechanics of the solution?

`cordova build android -- --minSdkVersion=15`

This build method has indeed fixed this issue but I'd like to understand why setting the `android-minSdkVersion` parameter in the `config.xml` is not the same as setting it as a command line argument. 

Also what is the importance of the empty `--` flag? The build fails without it but obviously it's hard/impossible to determine its intended usage.

**Comment by jeduan** [comment]
My understanding is that `--minSdkVersion` is not a parameter that `cordova-cli` understands but rather the Android compiler. The first `--` parameter specifies that everything following it should not be interpreted by the `cordova-cli` argument parser but passed directly to the programs it invokes.

**Comment by LukeeeeBennett** [comment]
Thanks @jeduan! Exactly what I was after.

**Comment by sagivf** [comment]
Does anybody use ionic? 
running ionic run build android -- --minSdkVersion=15 doesn't work.
Which means you can't debug and use live reload

**Comment by jeduan** [comment]
@sagivf that's an issue to raise on ionic-cli probably

**Comment by sagivf** [comment]
Good idea @jeduan, done.
https://github.com/driftyco/ionic-cli/issues/662

**Comment by mrinterweb** [comment]
For me @KeKs0r solution was what fixed the problem. Crosswalk was setting the minSdkVersion to 14. I just edited android-xwalk.gradle, and the problem was fixed, in my case.

**Comment by TanaseButcaru** [comment]
@jeduan solution worked for me. Used in combination with the latest version of crosswalk plugin (1.5.0).
``getConfigPreference`` function from ``cordova-plugin-crosswalk-webview`` reads the ``res/xml/config.xml`` file, not the ``AndroidManifest.xml``.
So, editing the root ``config.xml`` works :+1:

**Comment by Hirbod** [comment]
fixed with #161
